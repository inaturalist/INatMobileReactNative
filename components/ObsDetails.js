// @flow


import React from "react";
import { Text } from "react-native";
import type { Node } from "react";
import ViewWithFooter from "./SharedComponents/ViewWithFooter";
import { ScrollView } from "react-native-gesture-handler";

const ObsDetails = ( ): Node => (
  <ViewWithFooter>
    <ScrollView>
    <Text>username</Text>
    <Text>observation date</Text>
    <Text>observation photos, including faved observations & copywrite info</Text>
    <Text>default photo</Text>
    <Text>species card: rank, common name, sci name, same right hand side as obslist</Text>
    <Text>username</Text>
    <Text>location of obs</Text>
    <Text>two tabs: activity, data</Text>
    <Text>activity: username, computervision, leading/improving, time since comment</Text>
    <Text>default photo with common name/name</Text>
    <Text>comments</Text>
    <Text>agree button</Text>
    <Text>data: description, location map, date observed, date uploaded</Text>
    <Text>data: projects, annotations, obs fields, obs created using, copyright</Text>
    </ScrollView>
  </ViewWithFooter>
);

export default ObsDetails;
