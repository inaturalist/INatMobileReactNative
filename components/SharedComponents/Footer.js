// @flow strict-local

import * as React from "react";
import { Pressable, Text, View } from "react-native";
import { viewStyles } from "../../styles/sharedComponents/footer";

const Footer = ( ): React.Node => (
  <View style={[viewStyles.row, viewStyles.shadow]}>
    <Pressable>
      <Text>menu</Text>
    </Pressable>
    <Pressable>
      <Text>explore</Text>
    </Pressable>
    <Pressable>
      <Text>camera</Text>
    </Pressable>
    <Pressable>
      <Text>obs list</Text>
    </Pressable>
    <Pressable>
      <Text>notifications</Text>
    </Pressable>
  </View>
);

export default Footer;
