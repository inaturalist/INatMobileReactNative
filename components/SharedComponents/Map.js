// @flow

import * as React from "react";
import { View } from "react-native";
import MapView from "react-native-maps";
import { viewStyles } from "../../styles/sharedComponents/map";

type Props = {
  latitude: number,
  longitude: number
}

// this component will need a fallback to another map library
// for people who don't use GMaps (i.e. users in China)
const Map = ( { latitude, longitude }: Props ): React.Node => (
  <View style={viewStyles.mapContainer}>
    <MapView
      style={viewStyles.map}
      initialRegion={{
        latitude,
        longitude,
        latitudeDelta: 0.0922,
        longitudeDelta: 0.0421
      }}
    />
  </View>
);

export default Map;
