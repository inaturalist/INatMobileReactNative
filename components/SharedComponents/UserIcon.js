// @flow

import * as React from "react";
import { Image } from "react-native";

import { imageStyles } from "../../styles/obsDetails";

type Props = {
  uri: string,
  large?: boolean
}

const UserIcon = ( { uri, large }: Props ): React.Node => {
  const imageSource = { uri };

  return (
    <Image source={imageSource} style={[
      imageStyles.userProfileIcon,
      large && imageStyles.largeIcon
    ]} />
  );
};

export default UserIcon;
