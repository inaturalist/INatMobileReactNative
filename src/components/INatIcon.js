/**
 * INatIcon icon set component.
 * Usage: <INatIcon name="icon-name" size={20} color="#4F8EF7" />
 */

import createIconSet from "react-native-vector-icons/lib/create-icon-set";

const glyphMap = {
  "compass-rose": 57344
};

const iconSet = createIconSet( glyphMap, "inaturalisticons", "inaturalisticons.ttf" );

export default iconSet;
export const {
  Button,
  getImageSource,
  getImageSourceSync
} = iconSet;
