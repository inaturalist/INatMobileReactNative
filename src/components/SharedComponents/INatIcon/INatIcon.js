/**
 * INatIcon icon set component.
 * Usage: <INatIcon name="icon-name" size={20} color="#4F8EF7" />
 */

import createIconSet from "react-native-vector-icons/lib/create-icon-set";

const glyphMap = {
  "arrow-down": 57344,
  "arrow-left": 57345,
  "arrow-right": 57346,
  "arrow-up": 57347,
  "close-button-circle": 57348,
  "copyright-circle": 57349,
  "crop-photo-circle": 57350,
  "close-button-x": 57351,
  collapse: 57352,
  "compass-rose": 57354,
  expand: 57356,
  "flash-on": 57357,
  focus: 57358,
  "icn-captive": 57360,
  "icn-dna": 57361,
  identification: 57362,
  "icn-id-help": 57363,
  "icn-layers": 57364,
  "icn-leading-id": 57365,
  "icn-location-none": 57366,
  "icn-location-obscured": 57367,
  "icn-location-private": 57369,
  logomark: 57370,
  "photo-grid": 57371,
  "photo-quilt": 57372,
  "species-unknown": 57373,
  "iconic-amphibia": 57375,
  "iconic-animalia": 57376,
  "iconic-arachnida": 57377,
  "iconic-aves": 57378,
  "iconic-actinopterygii": 57379,
  "iconic-fungi": 57380,
  "iconic-insecta": 57381,
  "iconic-chromista": 57382,
  "iconic-mammalia": 57383,
  "iconic-mollusca": 57384,
  "iconic-plantae": 57385,
  "iconic-protozoa": 57386,
  "iconic-reptilia": 57387,
  "iconic-unknown": 57388,
  "icn-wild": 57389,
  camera: 57392,
  checkmark: 57393,
  chatbubble: 57396,
  comments: 57397,
  plus: 57398,
  trash: 57399,
  "star-outline": 57402,
  "triple-dots": 57404,
  "pen-and-paper": 57407,
  microphone: 57409,
  "identification-solid": 57411,
  "inat-wordmark": 57412,
  "ios-calendar-outline": 57413,
  people: 57414,
  "notifications-bell": 57417,
  treeview: 57418,
  pencil: 57420,
  "hamburger-menu": 57353,
  "flag-outline": 57403,
  "id-agree": 57359,
  "arrow-up-plain": 57368,
  "sparkly-label": 57355,
  flag: 57374,
  gallery: 57415,
  taxarank: 57390
};

const iconSet = createIconSet( glyphMap, "inaturalisticons", "inaturalisticons.ttf" );

export default iconSet;
export const {
  Button,
  getImageSource,
  getImageSourceSync
} = iconSet;

export {
  glyphMap
};
