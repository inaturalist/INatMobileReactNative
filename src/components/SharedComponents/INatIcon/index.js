// @flow

import type { Node } from "react";
import React from "react";
import { useTheme } from "react-native-paper";

import Icon from "./INatIcon";

// Flow types for the props
type Props = {
  name: string,
  color?: string,
  size?: number
};

// Most of these are names for these icons used in design mapped to more
// consistent and deduped filenames. We might also put aliases of convenience
// here, e.g. "speech" and "chat" might both map to "comments" if we find
// ourselves forgetting the name "comments"
const ALIASES = {
  addevidence: "plus",
  "evidence-add": "plus-bold",
  addid: "id-agree",
  back: "chevron-left",
  captive: "pot",
  checkmark: "check",
  "close-large": "close",
  "close-small": "close-bold",
  combine: "photos",
  multiplephotos: "photos",
  "multiplephotos-small": "photos-outline",
  "comment-fill": "comments",
  "comment-outline": "comments-outline",
  copyrightcc: "copyright",
  currentlocation: "location-crosshairs",
  cv: "sparkly-label",
  date: "clock-outline",
  delete: "trash-outline",
  "downvote-active": "arrow-down-bold-circle",
  "downvote-inactive": "arrow-down-bold-circle-outline",
  notes: "pencil-outline",
  edit: "pencil",
  "evidence-close": "close-bold",
  fail: "triangle-exclamation",
  "fave-active": "star",
  "fave-inactive": "star-bold-outline",
  filters: "sliders",
  flipcamera: "flip",
  geoprivacy: "globe",
  gridview: "grid-square",
  listview: "list-square",
  "id-large-fill": "label",
  "id-small-outline": "label-outline",
  journalposts: "book",
  location: "map-marker-outline",
  maplayers: "layers",
  multiplesounds: "sounds",
  notifications: "notifications-bell",
  observations: "binoculars",
  pass: "checkmark-circle",
  recordsound: "microphone",
  remark: "add-comment-outline",
  reset: "rotate-right",
  search: "magnifying-glass",
  separate: "grid",
  "sound-pause": "pause-circle",
  "sound-play": "play-circle",
  "sound-record": "microphone-circle",
  "sound-small": "sound-bold-outline",
  sound: "sound-outline",
  species: "leaf",
  sync: "rotate",
  "sync-unsynced": "rotate-exclamation",
  taxarankarrow: "info-circle-outline",
  "upload-arrow": "arrow-up",
  "upload-complete": "check",
  "upload-saved": "arrow-up-circle-dots",
  "upvote-active": "arrow-up-bold-circle",
  "upvote-inactive": "arrow-up-bold-circle-outline"
};

const INatIcon = ( { name, color, size }: Props ): Node => {
  const theme = useTheme();
  // Use default color if none is specified
  return <Icon name={ALIASES[name] || name} color={color || theme.colors.primary} size={size} />;
};

export default INatIcon;
