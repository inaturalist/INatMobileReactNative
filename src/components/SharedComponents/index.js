export { default as ActivityAnimation } from "./ActivityAnimation/ActivityAnimation";
export { default as ActivityCount } from "./ActivityCount/ActivityCount";
export { default as CommentsCount } from "./ActivityCount/CommentsCount";
export { default as IdentificationsCount } from "./ActivityCount/IdentificationsCount";
export { default as ActivityIndicator } from "./ActivityIndicator";
export { default as BackButton } from "./Buttons/BackButton";
export { default as Button } from "./Buttons/Button";
export { default as CloseButton } from "./Buttons/CloseButton";
export { default as EvidenceButton } from "./Buttons/EvidenceButton";
export { default as INatIconButton } from "./Buttons/INatIconButton";
export { default as RotatingINatIconButton } from "./Buttons/RotatingINatIconButton";
export { default as TransparentCircleButton } from "./Buttons/TransparentCircleButton";
export { default as Checkbox } from "./Checkbox";
export { default as ConfidenceInterval } from "./ConfidenceInterval";
export { default as DateDisplay } from "./DateDisplay";
export { default as DateTimePicker } from "./DateTimePicker";
export { default as DetailsMap } from "./DetailsMap";
export { default as DisplayTaxon } from "./DisplayTaxon";
export { default as DisplayTaxonName } from "./DisplayTaxonName";
export { default as Divider } from "./Divider/Divider";
export { default as FloatingActionBar } from "./FloatingActionBar";
export { default as HideView } from "./HideView";
export { default as IconicTaxonChooser } from "./IconicTaxonChooser";
export { default as IconicTaxonIcon } from "./IconicTaxonIcon";
export { default as INatIcon } from "./INatIcon";
export { default as InlineUser } from "./InlineUser/InlineUser";
export { default as InputField } from "./InputField";
export { default as KebabMenu } from "./KebabMenu";
export { default as LocationPermissionGate } from "./LocationPermissionGate";
export { default as Map } from "./Map";
export { default as MediaNavButtons } from "./MediaNavButtons";
export { default as Modal } from "./Modal";
export { default as Mortal } from "./Mortal";
export { default as ObservationLocation } from "./ObservationLocation";
export { default as ObservationsFlashList } from "./ObservationsFlashList/ObservationsFlashList";
export { default as ObsStatus } from "./ObsStatus";
export { default as OverviewCounts } from "./OverviewCounts";
export { default as PermissionGate } from "./PermissionGate";
export { default as PhotoCount } from "./PhotoCount";
export { default as ProjectListItem } from "./ProjectListItem";
export { default as QualityGradeStatus } from "./QualityGradeStatus/QualityGradeStatus";
export { default as RadioButtonRow } from "./RadioButtonRow";
export { default as ScrollableWithStickyHeader } from "./ScrollableWithStickyHeader";
export { default as ScrollViewWrapper } from "./ScrollViewWrapper";
export { default as SearchBar } from "./SearchBar";
export { default as BottomSheet } from "./Sheets/BottomSheet";
export { default as BottomSheetStandardBackdrop } from "./Sheets/BottomSheetStandardBackdrop";
export { default as PickerSheet } from "./Sheets/PickerSheet";
export { default as RadioButtonSheet } from "./Sheets/RadioButtonSheet";
export { default as TextInputSheet } from "./Sheets/TextInputSheet";
export { default as TextSheet } from "./Sheets/TextSheet";
export { default as WarningSheet } from "./Sheets/WarningSheet";
export { default as StickyToolbar } from "./StickyToolbar";
export { default as Tabs } from "./Tabs/Tabs";
export { default as TaxonResult } from "./TaxonResult";
export { default as Body1 } from "./Typography/Body1";
export { default as Body1Bold } from "./Typography/Body1Bold";
export { default as Body2 } from "./Typography/Body2";
export { default as Body3 } from "./Typography/Body3";
export { default as Body3Bold } from "./Typography/Body3Bold";
export { default as Body4 } from "./Typography/Body4";
export { default as Heading1 } from "./Typography/Heading1";
export { default as Heading2 } from "./Typography/Heading2";
export { default as Heading3 } from "./Typography/Heading3";
export { default as Heading4 } from "./Typography/Heading4";
export { default as Heading5 } from "./Typography/Heading5";
export { default as INatText } from "./Typography/INatText";
export { default as INatTextLight } from "./Typography/INatTextLight";
export { default as INatTextMedium } from "./Typography/INatTextMedium";
export { default as List1 } from "./Typography/List1";
export { default as List2 } from "./Typography/List2";
export { default as P } from "./Typography/P";
export { default as Subheading1 } from "./Typography/Subheading1";
export { default as UploadStatus } from "./UploadStatus/UploadStatus";
export { default as UserIcon } from "./UserIcon/UserIcon";
export { default as UserListItem } from "./UserListItem";
export { default as UserText } from "./UserText";
export { default as ViewWrapper } from "./ViewWrapper";
