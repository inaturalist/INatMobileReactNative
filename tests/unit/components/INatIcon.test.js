import { render, screen } from "@testing-library/react-native";
import { INatIcon } from "components/SharedComponents";
import { glyphMap } from "components/SharedComponents/INatIcon/INatIcon";
import React from "react";

const iconName = "camera";

describe( "INatIcon", () => {
  it( "renders correctly", () => {
    render(
      <INatIcon
        name={iconName}
        size={20}
        color="#4F8EF7"
      />
    );
    // Snapshot test
    expect( screen ).toMatchSnapshot();
  } );
} );

describe( "glyphMap", () => {
  it( "is an object", () => {
    expect( glyphMap ).toBeInstanceOf( Object );
  } );
  it( "has icon name as key", () => {
    expect( glyphMap ).toHaveProperty( iconName );
  } );
} );
