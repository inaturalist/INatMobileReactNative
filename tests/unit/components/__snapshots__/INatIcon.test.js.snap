// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INatIcon renders correctly 1`] = `
<Text
  allowFontScaling={false}
  selectable={false}
  style={
    Array [
      Object {
        "color": "#4F8EF7",
        "fontSize": 20,
      },
      undefined,
      Object {
        "fontFamily": "INatIcon",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ï„Œ
</Text>
`;
